File: javascript/core/src/engine/verbs/select.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import { all } from 'arquero'

import type { SelectArgs } from '../../types.js'
import { makeStepFunction, makeStepNode, wrapColumnStep } from '../factories.js'

const doSelect = wrapColumnStep<SelectArgs>((input, { columns = [] }) => {
	const expr = [columns] as any
	if (expr.length === 0) {
		expr.push(all())
	}
	return input.select(...expr)
})

export const select = makeStepFunction(doSelect)
export const selectNode = makeStepNode(doSelect)

File: python/datashaper/datashaper/engine/verbs/select.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Select verb implementation."""

from typing import Any

from datashaper.engine.verbs.verb_input import VerbInput
from datashaper.engine.verbs.verbs_mapping import verb
from datashaper.table_store.types import (
    VerbResult,
    create_verb_result,
)
from datashaper.verbs import select


@verb(name="select", treats_input_tables_as_immutable=True)
def select_verb(input: VerbInput, columns: list[str], **kwargs: Any) -> VerbResult:
    """Select verb implementation."""
    result = select(input.get_input(), columns, **kwargs)
    return create_verb_result(result)
